mono/metadata/assembly.c

```diff
@@ -1764,7 +1764,7 @@ mono_assembly_load_from_full (MonoImage *image, const char*fname,
 	 * assemblies lock.
 	 */
 	if (ass->aname.name) {
-		ass2 = mono_assembly_invoke_search_hook_internal (&ass->aname, refonly, FALSE);
+		ass2 = mono_assembly_invoke_search_hook_internal (&ass->aname, NULL, refonly, FALSE);
```

Comment:
See the test for .NET-framework's behavior. The event handler is called with `RequestingAssembly == null` for assemblies loaded with `AppDomain.Load` (or any of the other possible entry points). I'd guess the reasoning behind this is that explicitly loading does not mean requesting by an assembly, but requesting by user code.